### PR TITLE
[plugin.video.kla.tv] 0.0.4

### PR DESCRIPTION
**Pull request by celadoor**
### Description
Submission of new addon kla.tv
### Checklist:
- [X] My code follows the [add-on rules](http://kodi.wiki/view/Add-on_rules) and [piracy stance](http://kodi.wiki/view/Official:Forum_rules#Piracy_Policy) of this project. 
- [X] I have read the [CONTRIBUTING](https://github.com/xbmc/repo-plugins/blob/master/CONTRIBUTING.md) document
- [X] Each add-on submission should be a single commit with using the following style: [plugin.video.foo] v1.0.0

Additional information :
- Submitting your add-on to this specific branch makes it available to any Kodi version equal or higher than the branch name with the applicable Kodi dependencies limits.
- [add-on development](http://kodi.wiki/view/Add-on_development) wiki page.
- Kodi [pydocs](http://kodi.wiki/view/PyDocs) provide information about the Python API
- [PEP8](https://www.python.org/dev/peps/pep-0008/) codingstyle which is considered best practice but not mandatory.
- This add-on repository has automated code guideline check which could help you improve your coding. You can find the results of these check at [Codacy](https://www.codacy.com/app/Kodi/repo-plugins/dashboard). You can create your own account as well to continuously monitor your python coding before submitting to repo.
- Development questions can be asked in the [add-on development](http://forum.kodi.tv/forumdisplay.php?fid=26) section on the Kodi forum.

### CODE REVIEW

File: plugin.video.kla.tv/addon.py [hunk 1]
```diff
@@ -0,0 +1,169 @@
...
+try:
+    name = urllib.unquote_plus(params["name"])
+except:
+    pass
+try:
+    mode = int(params["mode"])
+except:
+    pass
+
+print "Mode: " + str(mode)
```
Comment (learningit) [inline plugin.video.kla.tv/addon.py]: Do not use print to log output. Please use xbmc.log() in its place.
Same in the many below this.

File: plugin.video.kla.tv/addon.xml [hunk 2]
```diff
@@ -0,0 +1,39 @@
+<?xml version='1.0' encoding='UTF-8' standalone='yes'?>
+<addon 
+  id="plugin.video.kla.tv" 
+  name="Klagemauer TV" 
+  provider-name="celadoor" 
+  version="0.0.4">
+  <requires>
+    <import  addon="xbmc.python"                      version="2.1.0" />
```
Comment (learningit) [inline plugin.video.kla.tv/addon.xml]: The next 3 imports are not used by your code, please remove them

File: plugin.video.kla.tv/addon.xml [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+    <description lang="pl">Z drugiej strony telewizor Klagenwand działa od 2012 roku jako ochotnik! I to w 90 studiach z ponad 170 prelegentami, w 38 językach i obszarze nadawczym 212 krajów. Pomóż, aby te ważne głosy słyszeć na całym świecie. W ciągu pierwszych dwóch lat Kla.TV wyprodukował ponad 5000 kanałów, w tym 30 filmów dokumentalnych i wiele innych..</description>
+    <description lang="fr">Klagenwand TV, en revanche, a travaillé comme bénévole pour vous depuis 2012! Et cela dans 90 studios avec plus de 170 présentateurs, en 38 langues et une zone de diffusion de 212 pays. Aidez à faire entendre ces voix importantes dans le monde entier. Au cours des deux premières années, Kla.TV a produit plus de 5 000 chaînes, dont 30 films documentaires et bien plus encore..</description>
+<news>Version 0.0.4 -Initial Release. Klagemauer TV video plugin.supports 4 languages de,en,pl,fr</news>
+    <platform>all</platform>
+    <language>en de pl fr</language>
+	<license>GNU GENERAL PUBLIC LICENSE. Version 3, June 2007</license>
+	<forum></forum>
+	<website>www.kla.tv</website>
+	<email></email>
+	<source></source>
```
Comment (learningit) [inline plugin.video.kla.tv/addon.xml]: please add your github source

File: plugin.video.kla.tv/addon.xml [hunk 4]
```diff
@@ -0,0 +1,39 @@
+<?xml version='1.0' encoding='UTF-8' standalone='yes'?>
+<addon 
+  id="plugin.video.kla.tv" 
+  name="Klagemauer TV" 
+  provider-name="celadoor" 
+  version="0.0.4">
+  <requires>
```
Comment (learningit) [inline plugin.video.kla.tv/addon.xml]: xbmc.python version should be '2.25.0' for Krypton

File: plugin.video.kla.tv/addon.py [hunk 5]
```diff
@@ -0,0 +1,169 @@
...
+    liz.setInfo(type="Video", infoLabels={"Title": name})
+    ok = xbmcplugin.addDirectoryItem(handle=int(sys.argv[1]), url=url, listitem=liz)
+    return ok
+
+
+def addDir(name, url, mode, iconimage):
+    u = sys.argv[0] + "?url=" + urllib.quote_plus(url) + "&mode=" + str(mode) + "&name=" + urllib.quote_plus(name)
+    ok = True
+    liz = xbmcgui.ListItem(name, iconImage="DefaultFolder.png", thumbnailImage=iconimage)
+    liz.setInfo(type="Video", infoLabels={"Title": name})
```
Comment (learningit) [inline plugin.video.kla.tv/addon.py]: Please set the 'mediatype' infoLabel to the correct value.

File: plugin.video.kla.tv/addon.py [hunk 6]
```diff
@@ -0,0 +1,169 @@
...
+            if (len(splitparams)) == 2:
+                param[splitparams[0]] = splitparams[1]
+
+    return param
+
+
+def addLink(name, url, iconimage):
+    ok = True
+    liz = xbmcgui.ListItem(name, iconImage="DefaultVideo.png", thumbnailImage=iconimage)
+    liz.setInfo(type="Video", infoLabels={"Title": name})
```
Comment (learningit) [inline plugin.video.kla.tv/addon.py]: Please set the 'mediatype' infoLabel to the correct value.

File: plugin.video.kla.tv/addon.xml [hunk 7]
```diff
@@ -0,0 +1,39 @@
...
+    <import  addon="xbmc.python"                      version="2.1.0" />
+    <import  addon="script.common.plugin.cache"       version="1.5.1" />
+    <import  addon="script.module.parsedom"           version="1.5.1" />
+    <import  addon="script.module.simple.downloader"  version="0.9.4" />
+  </requires>
+  <extension library="addon.py"                     point="xbmc.python.pluginsource">
+    <provides>video</provides>
+  </extension>
+  <extension point="xbmc.addon.metadata">
+    <summary lang="de">www.kla.tv video plugin</summary>
```
Comment (enen92) [inline plugin.video.kla.tv/addon.xml]: please also adjust the language tags to use "language-country" naming scheme. See here for more information: http://kodi.wiki/view/language_support

**Comment by celadoor** [comment]
Committed in accordance with review.

Should the language tag in `addon.xml` be adjusted to the same format as the `lang` attributes for `description` and `summary`? I have set them to be the same format as the language resource folders, let me know if there is a different format for these.

I have also set the `source` tag to be the plugin's folder on my fork, but I have noticed other plugins using their own independent repositories for this. Would it be better to create a repository for this, or is it fine as it is?

**Comment by enen92** [comment]
It is fine as it is. Thanks much for adjusting

**Pull request merged**